File: src/main/java/com/github/aivancioglo/resttest/setters/Protocol.java
package com.github.aivancioglo.resttest.setters;

import com.github.aivancioglo.resttest.http.HTTPRequest;

public class Protocol<T extends HTTPRequest> implements Setter<T> {
    private String protocol;

    public Protocol(String protocol) {
        this.protocol = protocol;
    }

    /**
     * Update request.
     * @param request actual request.
     */
    @Override
    public void update(T request) {
        request.setProtocol(protocol);
    }
}

File: src/main/kotlin/com/github/aivancioglo/resttest/setters/Protocol.kt
package com.github.aivancioglo.resttest.setters

import com.github.aivancioglo.resttest.http.HTTPRequest

/**
 * Protocol setter.
 *
 * @constructor is creating variable with requestSpecification protocol value.
 * @param protocol of your requestSpecification.
 */
class Protocol<in T : HTTPRequest<*>>(private val protocol: String) : Setter<T> {

    /**
     * Setting protocol of your requestSpecification.
     *
     * @param request that will be updated.
     */
    override fun update(request: T) {
        request.protocol = protocol
    }
}